GVRf/Framework/src/org/gearvrf/GVREventManager.java

```diff
@@ -0,0 +1,155 @@
+     * @param eventName
+     *     The name of the event, such as "onInit".
+     * @param params
+     *     Parameters of the event. It should match the parameter list
+     * of the corresponding method in the interface, specified by {@code
+     * event class}
+     */
+    public void sendEvent(Object target, Class<? extends IEvents> eventsClass,
+            String eventName, Object... params) {
+        // Validate the event
+        Method method = validateEvent(target, eventsClass, eventName, params);
+
+        // Try invoking the handler in the script
+        if (target instanceof IScriptable) {
+            tryInvokeScript((IScriptable)target, eventName, params);
+        }
+
+        // Try invoking the method in target
+        invokeMethod(target, method, params);
+    }
+
+    private Method validateEvent(Object target, Class<? extends IEvents> eventsClass,
+            String eventName, Object[] params) {
+        // Check target event interface
+        if (!eventsClass.isInstance(target)) {
+            throw new RuntimeException(String.format("The target object does not implement interface %s",
+                    eventsClass.getSimpleName()));
+        }
+
+        Method nameMatch = null;
+        for (Method method : eventsClass.getMethods()) {
+            // Match method name and event name
+            if (method.getName().equals(eventName)) {
+                nameMatch = method;
+
+                // Check number of parameters
+                Class<?>[] types = method.getParameterTypes();
+                if (types.length != params.length)
+                    continue;
+
+                // Check parameter types
+                int i = 0;
+                boolean foundMatchedMethod = true;
+                for (Class<?> type : types) {
+                    Object param = params[i++];
+                    if (!type.isInstance(param)) {
+                        foundMatchedMethod = false;
+                        break;
+                    }
+                }
+
+                if (foundMatchedMethod)
+                    return method;
+            }
+        }
+
+        // Error
+        if (nameMatch != null) {
+            throw new RuntimeException(String.format("The target object contains a method %s but parameters don't match",
+                    eventName));
+        } else {
+            throw new RuntimeException(String.format("The target object has no method %s",
+                    eventName));
+        }
+    }
+
+    private void tryInvokeScript(IScriptable target, String eventName,
+            Object[] params) {
+        GVRScriptFile script = mGvrContext.getScriptManager().getScriptFile(target);
+        if (script == null)
+            return;
+
+        script.invokeFunction(eventName, params);
+    }
+
+    private void invokeMethod(Object target, Method method, Object[] params) {
+        try {
+            method.invoke(target, params);
+        } catch (IllegalAccessException e) {
```

Comment:
invokeMethod is not designed to do anything more than just a method call. If the method throws, the caller should catch it.